service/src/main/java/bio/terra/catalog/service/dataset/Dataset.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.catalog.service.dataset;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public class Dataset {
```

Comment:
Can you use a java `record` for this? It would look something like
```
public record Dataset(UUID id, String datasetId, String storageSystem, String metadata, Instant createdDate) { }
```

Reply:
That looks so much better!